Headers/P7_Trace.h

```diff
@@ -72,14 +72,13 @@ enum eP7Trace_Level
 
 
 ////////////////////////////////////////////////////////////////////////////////
-#define P7_DELIVER(i_wID, i_eLevel, i_hModule, i_pFormat, ...) Trace(i_wID,\
-                                                                     i_eLevel,\
-                                                                     i_hModule,\
-                                                                     (tUINT16)__LINE__,\
-                                                                     __FILE__,\
-                                                                     __FUNCTION__,\
-                                                                     i_pFormat,\
-                                                                     __VA_ARGS__)
+#define P7_DELIVER(i_wID, i_eLevel, i_hModule, ...) Trace(  i_wID,\
+                                                            i_eLevel,\
+                                                            i_hModule,\
+                                                            (tUINT16)__LINE__,\
+                                                            __FILE__,\
+                                                            __FUNCTION__,\
+                                                            __VA_ARGS__)
```

Comment:
Yes, I did it intentionally, because, an expressions like:  LOG_ERROR(p7_modules::socket, "A message"); compiles with error. My compiler gcc expects more arguments after i_pFormat. It treats __VA_ARGS__ as "one or more arguments".

Now, I've modified the cpp test to demonstrate this. 098f2553b8da377c9cc755179583b34937888d43
This test doesn't compile on my side. The macro ##__VA_ARGS__  solves the problem, but you say it isn't crossplatform.

So, if you have any other idea how to solve the problem, I' ll be glad to see it.

Reply:
OK, got it